### DISCUSSION TITLE
Noise Reduction Off/On

### DISCUSSION THREAD

**Discussion by lyricsdude** [gopro/labs#973] [General]
Is it possible to turn on Noise Reduction only when the ISO is higher than 800? And off (NR01=1) when it's 800 or lower? :)

**Comment by dnewman-gpsw** [comment]
Most likely.  Try it. This is would a script that is checking the ISO, and enabling and disabling NR01 during capture.

**Reply by lyricsdude** [reply]
i have no idea how to do that tbh lol...

**Reply by dnewman-gpsw** [reply]
Something like ```=Ai=B0>A800=B1$NR01=B!1N!R```

![image](https://github.com/gopro/labs/assets/10856084/dac81b6f-9aed-42c7-8cad-902fd7cfa142)

Seems to work here.

**Reply by lyricsdude** [reply]
thank you :)

**Reply by lyricsdude** [reply]
i changed it to

=Ai=B1>A500=B2$NR01=B!1N!R

NR01=1 at all scenes but if ISO higher than 500 use NR01=2 to remove all the nasty tiny noise

Works fantastic on 5.3k! Thanks again

**Reply by dnewman-gpsw** [reply]
The deference between NR01=1 and NR01=2 is likely not perceivable, as those settings are noise reduction off and noise reduction at 2%.

**Reply by lyricsdude** [reply]
there's definitely a difference to me :) sharpening NR01=2 footage and then uploading 5.3k to youtube you hardly see any noise. But sharpening NR01=1 footage looks terrible

**Reply by dnewman-gpsw** [reply]
There is not sharpening difference between NR01=1 and NR01=2.  Here they are.  Can you pick which is which?
![NR01a](https://github.com/gopro/labs/assets/10856084/36924965-8f77-4293-bff3-074954503b75)
![NR01b](https://github.com/gopro/labs/assets/10856084/0202d6ef-770c-4dcb-9fbe-a729c8bdeedf)

**Reply by dnewman-gpsw** [reply]
It didn't have the BITR cranked up, so these just NR01 at 1 and 2 only.

**Reply by lyricsdude** [reply]
these are both NR01=1 as what I see, or it's not zoomed in enough. NR01=2 has a distinct pattern. That's why I started this thread before https://github.com/gopro/labs/discussions/535

Because I noticed that time that NR01=1 looked like NR01=2

**Reply by dnewman-gpsw** [reply]
They are what I said, I wasn't trying to trick you, one is NR01=1 the other =2 (zoom to 100% pixels, 1 to 1.)  #535 was fixed, and is it now very difficult tell, which you just proved thinking these are the same setting.  The noise reduction is between the cleaned and the original image, NR01 is just a percentage of that mix.

I run this script which will set NR01=0, then 1, 2, 3, etc. with one fast camera setup.  
```$NR01=A!2S!4E=A+1!R```  

Remember the longer the camera is on, the more thermal differences will affect the look of the noise.

**Reply by lyricsdude** [reply]
i'm looking at my footage now again from NR01=1 and NR01=2. I can clearly see the difference even from far. NR01=2 has that "Cheetah skin" pattern. And NR01=1 just has normal grainy noise.

**Reply by dnewman-gpsw** [reply]
Have you upgraded your firmware?  Did you try the above script?  I've tested this on HERO11 and HERO12, latest firmware, no sudden change between NR01=1 and 2 (or even 5).  Shooting 5.3K 16:9 24p, no stabilization, testing the 144th frame, so that HEVC has settled and that is an I-frame.

**Reply by lyricsdude** [reply]
I use 5.3k 60fps on the HERO11 latest firmware with stabilization. I will test myself if stabilization or framerate affects it

**Comment by Inter9800** [comment]
Hello everyone. Tell me, is it possible to embed in this qr code a command to change the dynamic range curve, for example logb=50? I apologize for the mistakes, the text was compiled with the help of an online translator

**Reply by dnewman-gpsw** [reply]
Yes, but ask in a new thread, as this is not related the noise reduction question.

**Reply by Inter9800** [reply]
> Yes, but ask in a new thread, as this is not related the noise reduction question.

It seems to me that the dynamic range curve is directly related to the amount of noise in the shadows, since the curve applies to all modes. And reducing the dynamic range gives, in my opinion, a very good result in low light

**Reply by dnewman-gpsw** [reply]
LOGB is not a noise control.  You can try it, although I believe a dynamic log curve would be harder to color correct.

**Reply by Inter9800** [reply]
> LOGB is not a noise control. You can try it, although I believe a dynamic log curve would be harder to color correct.

Logb seems to work for all modes in both standard and HDR, in my opinion, reducing to logb = 50 gives a very good result in low light. But in any case, thanks, I'll try to ask in a new topic.
